### PR TITLE
Resume work on unit/integration tests/codeCov

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
- move tests from test_client to library
- add code coverage
- add unit/IT tests via drone

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: src/androidTest/java/com/owncloud/android/AbstractIT.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.junit.runner.RunWith;
+
+/**
+ * Common base for all integration tests
+ */
+
+@RunWith(AndroidJUnit4.class)
+public abstract class AbstractIT {
+
+    static OwnCloudClient client;
```
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply src/androidTest/java/com/owncloud/android/AbstractIT.java]: Hm. What to do?
Codacy says "use explicit scoping" and Android Studio says "remove protected keyword and use package wide implicitly".
@AndyScherzinger

[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply src/androidTest/java/com/owncloud/android/AbstractIT.java]: ignore seems to be the right choice. I'll set it to ignore in codacy.

[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply src/androidTest/java/com/owncloud/android/AbstractIT.java]: :+1:

**Comment by AndyScherzinger** [comment]
👍 code looks good to me and CI seems to be happy too ;)

**Comment by tobiasKaminsky** [comment]
Codacy is way to slow today.
I'll merge now and if Codacy finds something, we can add it anyways…

**Comment by AndyScherzinger** [comment]
👍 perfectly fine with me!

**Pull request merged**